src/main/java/com/epam/test_generator/services/JiraService.java

```diff
@@ -0,0 +1,343 @@
+            suit.setPriority(getPriority(jiraStory.getPriority()));
+            suit.setLastModifiedDate(LocalDateTime.now());
+            suit.setLastJiraSyncDate(suit.getLastModifiedDate());
+            suit = suitDAO.save(suit);
+            Project project = projectDAO.findByJiraKey(projectKey);
+            checkNotNull(project);
+
+            project.getSuits().add(suit);
+        }
+    }
+
+    /**
+     * Creates and associates a new case in BDD to Jira sub story.
+     *
+     * @param jiraSubTask - new sub story from Jira
+     */
+    private void createCaseFromJiraSubTask(JiraSubTask jiraSubTask) {
+        Case caze = new Case();
+        caze.setName(jiraSubTask.getName());
+        caze.setDescription(jiraSubTask.getDescription());
+        caze.setPriority(getPriority(jiraSubTask.getPriority()));
+        caze.setJiraKey(jiraSubTask.getJiraKey());
+        caze.setJiraParentKey(jiraSubTask.getJiraParentKey());
+        caze.setJiraProjectKey(jiraSubTask.getJiraProjectKey());
+        caze.setLastModifiedDate(LocalDateTime.now());
+        caze.setLastJiraSyncDate(caze.getLastModifiedDate());
+
+        caseDAO.save(caze);
+
+        Suit suit = suitDAO.findByJiraKey(jiraSubTask.getJiraParentKey());
+        checkNotNull(suit);
+        suit.getCases().add(caze);
+    }
+
+
+    /**
+     * Updates existent suit in BDD by related Jira story.
+     *
+     * @param jiraStory - related issue from Jira
+     */
+    private void updateSuitFromJiraStory(JiraStory jiraStory) {
+        Suit suit = suitDAO.findByJiraKey(jiraStory.getJiraKey());
+        if (suit == null) {
+            return;
+        }
+        suit.setName(jiraStory.getName());
+        suit.setDescription(jiraStory.getDescription());
+        suit.setPriority(getPriority(jiraStory.getPriority()));
+        suit.setLastJiraSyncDate(LocalDateTime.now());
+
+        suitDAO.save(suit);
+    }
+
+
+    /**
+     * Updates existent case in BDD by related Jira sub story.
+     *
+     * @param jiraSubTask - related issue from Jira
+     */
+    private void updateSubTaskFromJiraSubTask(JiraSubTask jiraSubTask) {
+        Case caze = caseDAO.findByJiraKey(jiraSubTask.getJiraKey());
+        if (caze == null) {
+            return;
+        }
+        caze.setName(jiraSubTask.getName());
+        caze.setDescription(jiraSubTask.getDescription());
+        caze.setPriority(getPriority(jiraSubTask.getPriority()));
+        caze.setLastJiraSyncDate(LocalDateTime.now());
+        caseDAO.save(caze);
+    }
+
+    private Integer getPriority(String priority) {
+        if ("Critical".equals(priority)) {
+            return 1;
+        }
+        if ("Major".equals(priority)) {
+            return 2;
+        }
+        if ("Minor".equals(priority)) {
+            return 3;
+        }
+        if ("Trivial".equals(priority)) {
+            return 4;
+        }
+        if ("Blocker".equals(priority)) {
+            return 5;
+        }
+        return null;
+    }
```

Comment:
May be you could use switch-operator?
Look at this guide https://docs.oracle.com/javase/8/docs/technotes/guides/language/strings-switch.html

In addition, think is it convenient to use enums in such cases?
You could determine name and code fields in enum class and if- or case- blocks wouldn't be needed at all in that case.